docker-compose.yml

```diff
@@ -2,98 +2,168 @@ version: '2'
 
 services:
   config-server:
-    image: springcommunity/spring-petclinic-config-server
+    image: docker.io/springcommunity/spring-petclinic-config-server:2.7.5
     container_name: config-server
-    mem_limit: 512M
+    mem_limit: 768M
+    restart: on-failure
+    environment:
+      - SPRING_PROFILES_ACTIVE=docker
+    networks:
+      - default
     ports:
-     - 8888:8888
+      - 8888:8888
 
   discovery-server:
-    image: springcommunity/spring-petclinic-discovery-server
+    image: docker.io/springcommunity/spring-petclinic-discovery-server:2.7.5
     container_name: discovery-server
-    mem_limit: 512M
+    mem_limit: 768M
+    restart: on-failure
     depends_on:
-      - config-server
-    entrypoint: ["./dockerize","-wait=tcp://config-server:8888","-timeout=60s","--","java", "org.springframework.boot.loader.JarLauncher"]
+      config-server:
+        condition: service_started
+    environment:
+      - SPRING_PROFILES_ACTIVE=docker
+    networks:
+      - default
     ports:
-     - 8761:8761
+      - 8761:8761
 
   customers-service:
-    image: springcommunity/spring-petclinic-customers-service
+    image: docker.io/springcommunity/spring-petclinic-customers-service:2.7.5
     container_name: customers-service
-    mem_limit: 512M
+    mem_limit: 768M
+    restart: on-failure
     depends_on:
-     - config-server
-     - discovery-server
-    entrypoint: ["./dockerize","-wait=tcp://discovery-server:8761","-timeout=60s","--","java", "org.springframework.boot.loader.JarLauncher"]
+      config-server:
+        condition: service_started
+      discovery-server:
+        condition: service_started
+      sql-server:
+        condition: service_started
+    environment:
+      - SPRING_PROFILES_ACTIVE=docker,mysql
+    networks:
+      - default
     ports:
-    - 8081:8081
+      - 8081:8081
 
   visits-service:
-    image: springcommunity/spring-petclinic-visits-service
+    image: docker.io/springcommunity/spring-petclinic-visits-service:2.7.5
     container_name: visits-service
-    mem_limit: 512M
+    mem_limit: 768M
+    restart: on-failure
     depends_on:
-     - config-server
-     - discovery-server
-    entrypoint: ["./dockerize","-wait=tcp://discovery-server:8761","-timeout=60s","--","java", "org.springframework.boot.loader.JarLauncher"]
+      config-server:
+        condition: service_started
```

Comment:
With the use of the `service_started` condition (I didn't know it), do we really don't have to depend anymore from`dockerize`?
This will be a really good news :)

Reply:
Indeed, build is managed by Paketo (https://paketo.io), following Spring Boot recommendations (https://docs.spring.io/spring-boot/docs/current/maven-plugin/reference/htmlsingle/#build-image). The container starts by himself, startup command line is already written in the CMD/ENDTEYPOINT dockerfile build commands.

Reply:
I'm not sure that the `service_started` condition is sufficient. Like in k8s, the Spring Boot application may have started but is not ready to accept HTTP request.
According to the compose spec, the `service_healthy`  looks like better: https://github.com/compose-spec/compose-spec/blob/master/spec.md#healthcheck It requires that we implement healthcheck.
I found this sample: https://github.com/wpcfan/spring-boot-tut/blob/master/docker/docker-compose.yml